ocaml/parsing/parser.mly

```diff
@@ -153,7 +153,7 @@ let mk_attr ~loc name payload =
 (* For modes-related attributes, no need to call [register_attr] because they
 result from native syntax which is only parsed at proper places that are
 guaranteed to be used. *)
-let mkexp_with_modes ?(ghost=false) loc modes exp =
+let mkexp_with_modes ?(ghost=false) ~loc modes exp =
```

Comment:
I'm confused why `mkpat_with_modes` and `mktyp_with_modes` always call `Mode.ghostify`, but `mkexp_with_modes` makes its caller make the choice whether to call `Mode.ghostify`.
  
  * Can we always call `Mode.ghostify` in `mkexp_with_modes` as well?
  * Even simpler, can we just always make modes as ghost?

Reply:
In `mkexp_with_modes ?ghost ~loc modes exp`, the `ghost` doesn't control the ghost-ness of `modes`, but the whole application (applying `[%jane.erasable.mode modes]` to `exp`). The caller knows better if that should be ghost:
- If you write `local_ exp`, then I think the whole application should be non-ghost, and all modes should be non-ghost as well.
- If you write `let local_ a = exp`, it's parsed as:
```
let [@jane.erasable.mode "local"] a = [%jane.erasable.mode "local"] exp
```
The `pvb_attribute` will be non-ghost as well as all the modes in it. On the RHS, the whole application should be ghost, because it doesn't exist; All the modes should be ghost to avoid clash with the modes on the LHS.

Reply:
OK, I'm convinced enough.

Reply:
thanks - resolving.